File: week 1/happyNum.java
import java.util.HashSet;

class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> mySet = new HashSet();
        int val = 0, idx = 0;
        while (1 != 0)
        {
            val = 0;
            while (n != 0)
            {
                idx = n % 10;
                val += idx * idx;
                n /= 10;   
            }
            
            if (val == 1)
                return true;
            else if (mySet.contains(val))
                return false;
            
            mySet.add(val);
            n = val;
        }
        // return false;
    }
}

File: week 1/happyNum.cpp
#include <iostream>
#include <vector>
#include <set>
using namespace std;

bool isHappy(int n)
{
    set<int> mySet;
    int val;
    int idx;

    while (1)
    {
        val = 0;
        while (n)
        {
            idx = n % 10;
            val += idx * idx;
            n /= 10;
        }

        if (val == 1)
            return true;
        else if (mySet.find(val) != mySet.end())
            return false;

        mySet.insert(val);
        n = val;
    }
    return false;
}

int main(int args, char **argv)
{
    cout << (boolalpha) << isHappy(19) << endl;
    return 0;
}